### DISCUSSION TITLE
Some example to navigate search results?

### DISCUSSION THREAD

**Discussion by dirheim** [mozilla/pdf.js#18599] [Q&A]
I've managed to integrate pdfjs using the simple viewer example, but now I'm at a lose on how to navigate the results, calling the find again only resets the search, using the findcontroller doesn't seem to help.
I've tried to understand the web "example", it creates some events that dispatch events like "again", but I don't have any idea what are those events, how are they defined, I've tried to feed them to the eventBus with no luck.

**Comment by nicolo-ribaudo** [comment]
You can check [this class](https://github.com/mozilla/pdf.js/blob/aa2337f934d6e0b4ce9b8bfb45030fedbe3f01f4/web/pdf_find_bar.js#L97), which dispatches events for the viewer's find bar.

**Reply by dirheim** [reply]
This somehow helps, but it still doesn't help explain how that view example keeps tracks of which is the current match of the result.

**Reply by nicolo-ribaudo** [reply]
It's through the private `_offset` property in PDFFindController.

**Comment by dirheim** [comment]
_offset doesn't seem to give an absolute result, just a relative index to
the current page

[image: image.png]

For what it seems, I would need to do something like get selected.matchIdx
and add to the size of all idx from 0 to the current selected.pageIdx in
pageMatchesLength, but it surprises me that there is no simpler way to do it

On Tue, 13 Aug 2024 at 08:09, Nicolò Ribaudo ***@***.***>
wrote:

> It's through the private _offset property in PDFFindController.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/pdf.js/discussions/18599#discussioncomment-10321671>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGBHWXDXEFQGCN4DWDHEASTZRGPKXAVCNFSM6AAAAABML73LGKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZSGE3DOMI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
*Vicente Sampedro Burgos*
